### PR TITLE
Autodetect toolbox refactor in tsx

### PR DESCRIPTION
**Pull request by Park-J-lab**
AAD login is under debugging, other parts are good to go.

### CODE REVIEW

File: react-seed/src/components/Body.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { Component } from "react";
+import "../styles/App.css";
+import "antd/dist/antd.css";
+import "../styles/antd.css";
```
Comment (Wanjing-Li) [inline react-seed/src/components/Body.tsx:4]: What's the main purpose to import `antd.css` twice? 
Followed this [doc](https://ant.design/docs/react/use-with-create-react-app) to import it correctly.

File: react-seed/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "react-seed",
```
[the remark this answers is not in the document]
Comment (Wanjing-Li) [reply react-seed/package.json:2]: The root dir name  `react-seed` doesn't make sense to me.

[the remark this answers is not in the document]
Comment (haven2world) [reply react-seed/package.json:2]: I think it'd be better to move everything in `react-seed` to the root dir

File: react-seed/src/components/Response.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  response: any;
+  loading: boolean;
+}) => {
+  const { loading, response } = props;
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+  return (
+    <div className="JSONResponse">
```
Comment (haven2world) [inline react-seed/src/components/Response.tsx:16]: How about JSON-response

File: react-seed/src/components/SearchBox.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import React, { Component } from "react";
+import "../styles/App.css";
```
Comment (haven2world) [inline react-seed/src/components/SearchBox.tsx:2]: I think they only need to be imported once in `App.tsx`. Does it work?

File: react-seed/src/pages/Home/AutoDetect.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      loading: false,
+      visible: false,
+      defaultKey: "1",
+    };
+  }
+
+  getResponse: (email: string) => Promise<void> = async (email: string) => {
+    this.setState({ loading: true });
+
+    const res = http.get<AxiosResponse>(
```
Comment (haven2world) [inline react-seed/src/pages/Home/AutoDetect.tsx:32]: Could we move this to Service.ts?

File: react-seed/src/pages/Home/AutoDiscoverExo.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      loading: false,
+      visible: false,
+      defaultKey: "4",
+    };
+  }
+
+  getResponse: (email: string) => Promise<void> = async (email: string) => {
+    this.setState({ loading: true });
+
+    const res = http.get<AxiosResponse>(
```
Comment (haven2world) [inline react-seed/src/pages/Home/AutoDiscoverExo.tsx:32]: ditto

File: react-seed/src/pages/Home/Home.css [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+.App-header {
+  background-color: #282c34;
+  min-height: 100vh;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
```
Comment (haven2world) [inline react-seed/src/pages/Home/Home.css:23]: Why we use this font size?

File: react-seed/src/pages/Home/Home.test.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (haven2world) [inline react-seed/src/pages/Home/Home.test.tsx:1]: Could we removed this file?

File: react-seed/src/pages/Login/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+              <div className="myitem-1" style={{ marginLeft: "1800%" }}>
+                {isAuthenticated ? undefined : <SignInButton />}
+              </div>
+            </div>
+          </Col>
+        </Row>
+      </div>
+      <div>
+        <div
+          style={{
```
Comment (haven2world) [inline react-seed/src/pages/Login/Login.tsx:31]: Let's move all styles in this file to a css file

File: react-seed/src/mock/fetchWfhStatusMock.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
+export const fetchWfhStatusMock = [{
```
Comment (schuberng) [inline react-seed/src/mock/fetchWfhStatusMock.ts:1]: I don't think this Mock file is used in our project, is it?

File: react-seed/src/util/authConfig.ts [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        message: string,
+        containsPii: boolean,
+      ): void => {
+        if (containsPii) {
+          return;
+        }
+        switch (level) {
+          default:
+          case LogLevel.Error:
+            console.error(message);
```
Comment (schuberng) [inline react-seed/src/util/authConfig.ts:37]: Do we need console.log to find error info?

File: react-seed/src/util/utils.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import { IPublicClientApplication } from "@azure/msal-browser";
+import { loginRequest } from "./authConfig";
+
+export function RenderIf(flag: boolean): ((_: React.ReactNode) => React.ReactNode | null) {
+  return (viewContent: React.ReactNode): React.ReactNode | null => (flag ? viewContent : null);
+}
+
+export const handleLogout = (instance: IPublicClientApplication): void => {
+  instance.logoutRedirect().catch((e: any) => {
+    console.error(e);
```
Comment (schuberng) [inline react-seed/src/util/utils.ts:11]: same here

**Review by schuberng — CHANGES_REQUESTED** [review]
Please resolve all comments before merge this PR.

**Review by schuberng — CHANGES_REQUESTED** [review]

File: react-seed/src/Router.tsx [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Route, BrowserRouter } from "react-router-dom";
+import { Autodetect } from "./pages/Home/AutoDetect";
+import { AutoDiscoverExo } from "./pages/Home/AutoDiscoverExo";
+import { AutoDiscoverOnPrem } from "./pages/Home/AutoDiscoverOnPrem";
+import { FetchEas } from "./pages/Home/FetchEas";
+import { SovereignDiscover } from "./pages/Home/SovereignDiscover";
+
+export const RootRouter = (): JSX.Element => (
```
Comment (Wanjing-Li) [inline react-seed/src/Router.tsx:9]: Can  you change the return type as `React.FC<props>`?

File: react-seed/src/components/Menu.tsx [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+      >
+        Sign out
+      </Button>
+    </Menu.Item>
+  </Menu>
+);
+
+export const LoadingMenu: (instance: any) => JSX.Element = (instance: any) => (
+  <Menu>
+    <Menu.Item key="0">Loading</Menu.Item>
```
Comment (Wanjing-Li) [inline react-seed/src/components/Menu.tsx:31]: What's the difference between `ContentMenu` and `LodingMenu`?  Can you pass these parameters as config and use `Factory Pattern` to generate the menu?

File: react-seed/src/components/RequestDisplayer.tsx [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { Card } from "antd";
+import React from "react";
+import * as service from "../util/service";
+
+export const RequestDisplayer: React.FC<{
+  requestResponse: service.FetchWfhStatusRes;
```
Comment (Wanjing-Li) [inline react-seed/src/components/RequestDisplayer.tsx:6]: Define this as a type Props?

File: react-seed/src/components/Response.tsx [hunk 16]
```diff
@@ -0,0 +1,24 @@
+import PropTypes from "prop-types";
+import "../styles/App.css";
```
Comment (Wanjing-Li) [inline react-seed/src/components/Response.tsx:2]: `App.css` has been imported into `App.tsx`

File: react-seed/src/components/SearchBox.tsx [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+  text: string;
+};
+
+export class Searchbox extends Component<SearchProps, SearchState> {
+  constructor(props: SearchProps | Readonly<SearchProps>) {
+    super(props);
+    this.state = { text: "" };
+  }
+
+  onSubmit = (e: { preventDefault: () => void }): void => {
```
Comment (Wanjing-Li) [inline react-seed/src/components/SearchBox.tsx:23]: I think the type of `e` shouldn't be this. How about `React.xxxEvent<xxxElement>`?

File: react-seed/src/components/SearchBox.tsx [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+    e.preventDefault();
+    this.onSearch();
+  };
+
+  onChange = (e: { target: { name: string; value: string } }): void => {
```
Comment (Wanjing-Li) [inline react-seed/src/components/SearchBox.tsx:28]: same here

File: react-seed/src/components/SearchBox.tsx [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    getResponse(text);
+  };
+
+  render(): JSX.Element {
+    const { text }: SearchState = this.state;
+
+    return (
+      <div>
+        <Search
+          className="Searchbar"
```
Comment (Wanjing-Li) [inline react-seed/src/components/SearchBox.tsx:44]: camel-case?

File: react-seed/src/components/SignOutButton.tsx [hunk 20]
```diff
@@ -0,0 +1,23 @@
+import { useMsal } from "@azure/msal-react";
+import Button from "react-bootstrap/Button";
+import { handleLogout } from "../util/utils";
+
+export const SignOutButton = () => {
```
Comment (Wanjing-Li) [inline react-seed/src/components/SignOutButton.tsx:5]: The DOM structure of `sign in` and `sign out` looks the same, merge them into one component?

File: react-seed/src/pages/Home/AutoDetect.tsx [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+type PageProps = {}
+
+type PageState = {
+  response: any;
+  loading: boolean;
+  visible: boolean;
+  defaultKey: string;
+};
+
+export class Autodetect extends Component<PageProps, PageState> {
```
Comment (Wanjing-Li) [inline react-seed/src/pages/Home/AutoDetect.tsx:18]: Merge AutoDetect and AutoDiscoverExo and AutoDiscoverOnPrem and FetchEas and SovereignDiscover?

File: react-seed/src/styles/App.css [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+  text-align: left;
+  width: 100%;
+}
+
+/* My Navbar */
+.mynavbar {
+  display: flex;
+  justify-content: space-between;
+  align-items: left;
+  padding: 1rem 1.25rem;
```
Comment (Wanjing-Li) [inline react-seed/src/styles/App.css:134]: Why your css rules contain relative and absolute units ? `px, em, vh, vmin, %`

File: react-seed/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/package.json:2]: Resolved.

File: react-seed/src/Router.tsx [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+  <BrowserRouter>
+    <Route path="/autodetect" exact component={Autodetect} />
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/Router.tsx:11]: Modified.

File: react-seed/src/components/Body.tsx [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+};
+
+export class Body extends Component<BodyProps, BodyState> {
+  constructor(props: BodyProps | Readonly<BodyProps>) {
+    super(props);
+    this.state = { collapsed: false };
+  }
+
+  onCollapse = (collapsed: boolean): void => {
+    console.log(collapsed);
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/components/Body.tsx:37]: Removed.

File: react-seed/src/components/Body.tsx [hunk 25]
```diff
@@ -0,0 +1,97 @@
...
+          <Menu
+            theme="light"
+            defaultSelectedKeys={[defaultKey]}
+            defaultOpenKeys={[defaultKey]}
+            mode="inline"
+            subMenuCloseDelay={0.1}
+            triggerSubMenuAction="hover"
+          >
+            <Menu.Item key="1" icon={<PieChartOutlined />}>
+              <a href="/autodetect">AutoDetect</a>
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/components/Body.tsx:61]: Thanks, I have replaced.

File: react-seed/src/components/Navbar.tsx [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+import "../styles/App.css";
+import "antd/dist/antd.css";
+
+export const Navbar: React.FC = () => (
+  <div className="navbar">
+    <Row justify="space-between">
+      <Col className="MicrosoftLogo" xs={1} style={{ marginLeft: "0px" }}>
+        <img src={icon} alt="" />
+      </Col>
+      <Col span={1} style={{ marginLeft: "6%" }}>
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/components/Navbar.tsx:14]: The purpose is just to place this element in a proper position to make the whole part coordinated.

File: react-seed/src/components/Navbar.tsx [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+        <div className="container">
+          <div className="myitem-0">
+            <a href="/">Autodetect Toolbox</a>
+          </div>
+          <div className="myitem-1" style={{ marginLeft: "1800%" }}>
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/components/Navbar.tsx:19]: Removed.

File: react-seed/src/components/ProfileContent.tsx [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+  const name = accounts ? accounts[0] && accounts[0].name : "User";
+  const email = accounts[0] && accounts[0].username;
+
+  const nameArr = name!.toUpperCase().split(" ");
+  const nameLen = nameArr.length;
+  const firstName = nameArr[0];
+  const lastName = name === "User" || nameLen === 1 ? "" : nameArr[nameLen - 1];
+
+  return (
+    <div style={{ marginTop: "-5%" }}>
```
[the remark this answers is not in the document]
Comment (Park-J-lab) [reply react-seed/src/components/ProfileContent.tsx:19]: Removed.

File: react-seed/src/components/Response.tsx (hunk 3 above)
Comment (Park-J-lab) [reply react-seed/src/components/Response.tsx:16]: Good idea.

File: react-seed/src/pages/Home/AutoDetect.tsx (hunk 5 above)
Comment (Park-J-lab) [reply react-seed/src/pages/Home/AutoDetect.tsx:32]: Good idea. Resolved.

File: react-seed/src/pages/Home/AutoDiscoverExo.tsx (hunk 6 above)
Comment (Park-J-lab) [reply react-seed/src/pages/Home/AutoDiscoverExo.tsx:32]: Resolved.

File: react-seed/src/util/authConfig.ts (hunk 11 above)
Comment (Park-J-lab) [reply react-seed/src/util/authConfig.ts]: Removed.

File: react-seed/src/components/RequestDisplayer.tsx (hunk 15 above)
Comment (Park-J-lab) [reply react-seed/src/components/RequestDisplayer.tsx]: Removed this file.

File: react-seed/src/pages/Home/AutoDetect.tsx (hunk 21 above)
Comment (Park-J-lab) [reply react-seed/src/pages/Home/AutoDetect.tsx]: I extracted some public components.

File: react-seed/src/components/Body.tsx (hunk 1 above)
Comment (Park-J-lab) [reply react-seed/src/components/Body.tsx]: resolved

File: react-seed/src/components/Menu.tsx (hunk 14 above)
Comment (Park-J-lab) [reply react-seed/src/components/Menu.tsx]: The loadingmenu is quite different from contentmenu.

File: react-seed/src/components/Response.tsx (hunk 16 above)
Comment (Park-J-lab) [reply react-seed/src/components/Response.tsx]: Resolved.

File: react-seed/src/components/SearchBox.tsx (hunk 19 above)
Comment (Park-J-lab) [reply react-seed/src/components/SearchBox.tsx]: Resolved.

File: react-seed/src/pages/Home/Home.css (hunk 7 above)
Comment (Park-J-lab) [reply react-seed/src/pages/Home/Home.css]: Removed.

File: react-seed/src/components/SignOutButton.tsx (hunk 20 above)
Comment (Park-J-lab) [reply react-seed/src/components/SignOutButton.tsx]: Good idea, but it is not easy to merge.

File: react-seed/src/styles/App.css (hunk 22 above)
Comment (Park-J-lab) [reply react-seed/src/styles/App.css]: This is for specific appearance.

**Pull request closed without merging**